destination/writer/writer.go

```diff
@@ -38,25 +38,28 @@ const (
 
 // Writer implements a writer logic for ClickHouse destination.
 type Writer struct {
-	db          *sqlx.DB
-	table       string
-	keyColumns  []string
-	columnTypes map[string]string
+	db                 *sqlx.DB
+	table              string
+	keyColumns         []string
+	isSupportMutations bool
+	columnTypes        map[string]string
 }
 
 // Params is an incoming params for the New function.
 type Params struct {
-	DB         *sqlx.DB
-	Table      string
-	KeyColumns []string
+	DB                 *sqlx.DB
+	Table              string
+	KeyColumns         []string
+	IsSupportMutations bool
```

Comment:
Nitpick: I'd rename `IsSupportMutations` to `SupportsMutations`. Or reverse the logic and name it `CreatesOnly`.